keel-orca/src/main/kotlin/com/netflix/spinnaker/keel/orca/OrcaTaskMonitorAgent.kt

```diff
@@ -0,0 +1,90 @@
+package com.netflix.spinnaker.keel.orca
+
+import com.fasterxml.jackson.module.kotlin.convertValue
+import com.netflix.spinnaker.keel.api.ResourceId
+import com.netflix.spinnaker.keel.api.SubjectType
+import com.netflix.spinnaker.keel.events.ResourceTaskFailed
+import com.netflix.spinnaker.keel.events.ResourceTaskSucceeded
+import com.netflix.spinnaker.keel.events.TaskCreatedEvent
+import com.netflix.spinnaker.keel.persistence.NoSuchResourceId
+import com.netflix.spinnaker.keel.persistence.ResourceRepository
+import com.netflix.spinnaker.keel.persistence.TaskTrackingRepository
+import com.netflix.spinnaker.keel.scheduled.ScheduledAgent
+import com.netflix.spinnaker.keel.serialization.configuredObjectMapper
+import java.time.Clock
+import java.util.concurrent.TimeUnit
+import kotlinx.coroutines.async
+import kotlinx.coroutines.coroutineScope
+import org.slf4j.LoggerFactory
+import org.springframework.context.ApplicationEventPublisher
+import org.springframework.context.event.EventListener
+import org.springframework.stereotype.Component
+
+@Component
+class OrcaTaskMonitorAgent(
+  private val taskTrackingRepository: TaskTrackingRepository,
+  private val resourceRepository: ResourceRepository,
+  private val orcaService: OrcaService,
+  private val publisher: ApplicationEventPublisher,
+  private val clock: Clock
+) : ScheduledAgent {
+
+  private val log by lazy { LoggerFactory.getLogger(javaClass) }
+
+  override val lockTimeoutSeconds = TimeUnit.MINUTES.toSeconds(1)
+
+  private val mapper = configuredObjectMapper()
+
+  @EventListener(TaskCreatedEvent::class)
+  fun onTaskEvent(event: TaskCreatedEvent) {
+    taskTrackingRepository.store(event.taskRecord)
+  }
+
+  // 1. Get active tasks from task tracking table
+  // 2. For each task, call orca and ask for details
+  // 3. For each completed task, will emit an event for success/failure
+  override suspend fun invokeAgent() {
+    coroutineScope {
+      taskTrackingRepository.getTasks()
```

Comment:
This doesn't need to be addressed in this PR, but we should be sure to limit the number of tasks we'll concurrently request from orca here. Since `getTasks()` returns a list, can use `List.chunked()` to chunk the task list into smaller lists, which we'd use to put a ceiling on the number of concurrent orca requests we'd `await` for.